pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/UpdateKitToLegacyIdService.java

```diff
@@ -105,18 +107,29 @@ protected static UpdateLog changeKitIdsToLegacyIds(UpdateKitToLegacyIdsRequest u
                     + " will not update kit %s").formatted(updateKitToLegacyIdsRequest.getLegacyShortId(), legacyShortIdOnFile, shortId,
                     currentCollaboratorSampleId));
         }
-        if (!existsKitRequestWithCollaboratorSampleId(currentCollaboratorSampleId, profile.getGuid(), kitRequestDao)) {
+        if (!existsKitRequestWithCollaboratorSampleId(currentCollaboratorSampleId, kitRequestDao)) {
             return new UpdateLog(shortId, UpdateLog.UpdateStatus.ERROR,
                     ("Kit request not found for collaborator sample ID %s".formatted(currentCollaboratorSampleId)));
         }
-        if (existsKitRequestWithCollaboratorSampleId(newCollaboratorSampleId, profile.getGuid(), kitRequestDao)) {
+        if (existsKitRequestWithCollaboratorSampleId(newCollaboratorSampleId, kitRequestDao)) {
             return new UpdateLog(shortId, UpdateLog.UpdateStatus.ERROR,
                     "Kit request with the new collaboratorSampleId %s already exists!".formatted(newCollaboratorSampleId));
         }
         try {
             String ddpParticipantId = profile.getGuid();
             kitRequestDao.updateKitToLegacyIds(updateKitToLegacyIdsRequest, legacyParticipantId);
-            changeDataInEs(currentCollaboratorSampleId, newCollaboratorSampleId, ddpParticipantId, ddpInstanceDto.getEsParticipantIndex());
+            //find the recently updated kit request, so we can update the ES
+            List<KitRequestShipping> kitsWithNewSampleId = kitRequestDao.getKitRequestForCollaboratorSampleId(newCollaboratorSampleId);
+            if (kitsWithNewSampleId.size() != 1) {
```

Comment:
This new code block probably belongs inside `changeDataInEs` since none of the data is used in this method.